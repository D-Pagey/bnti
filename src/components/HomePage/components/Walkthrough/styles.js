import styled from 'styled-components';
import { margin, padding } from 'polished';
import { colour, fontSize, spacing } from '../../../../tokens';

export const Background = styled.div`
    background-color: ${colour.blue};
    ${padding(spacing.large, spacing.none)}
    width: 100%;
`;

export const Wrapper = styled.div`
    display: flex;
    justify-content: center;
    margin: ${spacing.big} auto;
`;

export const Step = styled.div`
    align-items: center;
    border-right: 1px solid ${colour.black};
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    max-width: 225px;
    ${padding(spacing.medium, spacing.medium, spacing.big)}
    width: 25%;

    &:last-child {
        border: 0;
    }
`;

export const Image = styled.img`
    height: auto;
    width: ${({ width = '100px' }) => width};
`;

export const Subtitle = styled.h3`
    font-size: ${fontSize.large};
    ${margin(spacing.medium, spacing.none)}
    text-align: center;
`;

export const Text = styled.p`
    margin: ${spacing.none};
    text-align: center;
`;
