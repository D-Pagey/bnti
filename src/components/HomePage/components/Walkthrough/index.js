import React from 'react';
import calendar from './assets/calendar.png';
import loot from './assets/loot.png';
import multiChips from './assets/3chips.png';
import chip from './assets/chip.png';
import * as S from './styles';

const Walkthrough = () => {
    return (
        <S.Background>
            <S.Wrapper>
                <S.Step>
                    <S.Image src={calendar} alt="calendar" />
                    <div>
                        <S.Subtitle>Step 1</S.Subtitle>
                        <S.Text>Find your tournament, you can filter by game and date.</S.Text>
                    </div>
                </S.Step>
                <S.Step>
                    <S.Image src={chip} alt="calendar" />
                    <div>
                        <S.Subtitle>Step 2</S.Subtitle>
                        <S.Text>Place your bounty to enter your tournament</S.Text>
                    </div>
                </S.Step>
                <S.Step>
                    <S.Image src={loot} alt="calendar" width="160px" />
                    <div>
                        <S.Subtitle>Step 3</S.Subtitle>
                        <S.Text>Kill players to earn their bounties, tracked by BRB.</S.Text>
                    </div>
                </S.Step>
                <S.Step>
                    <S.Image src={multiChips} alt="calendar" />
                    <div>
                        <S.Subtitle>Step 4</S.Subtitle>
                        <S.Text>Cash in your bounties. Gloat (optional)</S.Text>
                    </div>
                </S.Step>
            </S.Wrapper>
        </S.Background>
    );
};

export default Walkthrough;
