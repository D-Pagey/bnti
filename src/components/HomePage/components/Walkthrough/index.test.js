import React from 'react';
import { render } from 'react-testing-library';
import Walkthrough from '.';

describe('Walkthrough component', () => {
    it('should render', () => {
        const { container } = render(<Walkthrough />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
