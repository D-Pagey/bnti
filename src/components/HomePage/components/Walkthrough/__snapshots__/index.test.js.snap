// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Walkthrough component should render 1`] = `
.c0 {
  background-color: #00A7CB;
  padding-top: 1.5rem;
  padding-right: 0rem;
  padding-bottom: 1.5rem;
  padding-left: 0rem;
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 2rem auto;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-right: 1px solid #353535;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  max-width: 225px;
  padding-top: 1rem;
  padding-right: 1rem;
  padding-bottom: 2rem;
  padding-left: 1rem;
  width: 25%;
}

.c2:last-child {
  border: 0;
}

.c3 {
  height: auto;
  width: 100px;
}

.c6 {
  height: auto;
  width: 160px;
}

.c4 {
  font-size: 1.5rem;
  margin-top: 1rem;
  margin-right: 0rem;
  margin-bottom: 1rem;
  margin-left: 0rem;
  text-align: center;
}

.c5 {
  margin: 0rem;
  text-align: center;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <img
        alt="calendar"
        class="c3"
        src="calendar.png"
      />
      <div>
        <h3
          class="c4"
        >
          Step 1
        </h3>
        <p
          class="c5"
        >
          Find your tournament, you can filter by game and date.
        </p>
      </div>
    </div>
    <div
      class="c2"
    >
      <img
        alt="calendar"
        class="c3"
        src="chip.png"
      />
      <div>
        <h3
          class="c4"
        >
          Step 2
        </h3>
        <p
          class="c5"
        >
          Place your bounty to enter your tournament
        </p>
      </div>
    </div>
    <div
      class="c2"
    >
      <img
        alt="calendar"
        class="c6"
        src="loot.png"
        width="160px"
      />
      <div>
        <h3
          class="c4"
        >
          Step 3
        </h3>
        <p
          class="c5"
        >
          Kill players to earn their bounties, tracked by BRB.
        </p>
      </div>
    </div>
    <div
      class="c2"
    >
      <img
        alt="calendar"
        class="c3"
        src="3chips.png"
      />
      <div>
        <h3
          class="c4"
        >
          Step 4
        </h3>
        <p
          class="c5"
        >
          Cash in your bounties. Gloat (optional)
        </p>
      </div>
    </div>
  </div>
</div>
`;
